File: backend_tesis/codigos_fuente/usuario_1/proyecto_1/Perro.java
public class Perro extends Animal {

    // --- Atributos Propios ---
    private boolean ladrar;

    // --- Constructores ---
    public Perro() {
        super(); // Llama al constructor del padre
    }

    public Perro(int age, String gender, boolean ladrar) {
        super(age, gender); // Inicializa atributos del padre
        this.ladrar = ladrar;
    }

    // --- Métodos Propios ---
    private void correrRapido() {


        System.out.println("Perro corriendo");    
    
    
    }

    // --- Métodos heredados de Animal (Ejemplo de sobreescritura) ---
    @Override
    public void isFamele() {
        // Puedes agregar lógica extra aquí
        super.isFamele(); // Llama a la versión del padre
    }

}

File: backend_tesis/codigos_fuente/usuario_1/proyecto_17/Perro.cpp
#include <iostream>
#include <string>
using namespace std;

class Perro : public Animal {
public:
    int test;
// --- Constructores ---
    Perro() : Animal() {}
    Perro(string nombre, int test) : Animal(nombre), test(test) {}

    // --- Sobreescritura de Animal ---
    void hacerSonido() override {
        // Lógica de sobreescritura
        Animal::hacerSonido();
    }

};